scripts/defaultCIBuild.cmake

```diff
@@ -0,0 +1,30 @@
+#[[
+
+Default CMake cache for building in CI with most options enabled and a sane
+default value for CUDA architectures.
+
+#]]
+message(STATUS "Enabling both shared and static libraries")
+set(HIOP_BUILD_SHARED ON CACHE BOOL "")
+set(HIOP_BUILD_STATIC ON CACHE BOOL "")
+
+set(ENABLE_TESTS ON CACHE BOOL "")
+
+message(STATUS "Enabling MPI, RAJA, Umpire, Kron Reduction, and CUDA")
+set(HIOP_USE_MPI ON CACHE BOOL "")
+set(HIOP_USE_RAJA ON CACHE BOOL "")
+set(HIOP_USE_UMPIRE ON CACHE BOOL "")
+set(HIOP_WITH_KRON_REDUCTION ON CACHE BOOL "")
+set(HIOP_USE_GPU ON CACHE BOOL "")
+set(HIOP_USE_CUDA ON CACHE BOOL "")
+
+message(STATUS "Enabling HiOp's Sparse Interface")
+set(HIOP_SPARSE ON CACHE BOOL "")
+
+message(STATUS "Enabling HiOp's deepchecking")
+set(HIOP_DEEPCHECKS ON CACHE BOOL "")
+
+message(STATUS "Setting default cuda architecture to 60")
+set(CMAKE_CUDA_ARCHITECTURES 60 CACHE STRING "")
```

Comment:
Meant to leave this on the review, but something like this should work:
```CMake
if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
  message(STATUS "Setting default cuda architecture to 70")
  set(CMAKE_CUDA_ARCHITECTURES 70 CACHE STRING "")
else()
  message(STATUS "Using cuda architecture ${CMAKE_CUDA_ARCHITECTURES} from cache")
endif()
```

Reply:
Refer to my comment above - just adding `-DCMAKE_CUDA_ARCHITECTURES=XX` will override whatever I have in this cache script